rpc/handlers.go

```diff
@@ -1263,6 +1207,90 @@ func (h *Handler) SimulateTransactions(id BlockID, transactions []BroadcastedTra
 	return result, nil
 }
 
+func (h *Handler) TraceBlockTransactions(blockHash felt.Felt) ([]TracedBlockTransaction, *jsonrpc.Error) {
+	block, err := h.bcReader.BlockByHash(&blockHash)
+	if err != nil {
+		return nil, ErrBlockNotFound
+	}
+
+	return h.traceBlockTransactions(block)
+}
+
+func (h *Handler) traceBlockTransactions(block *core.Block) ([]TracedBlockTransaction, *jsonrpc.Error) {
```

Comment:
there is one flaw with this approach: traceTransaction will execute whole block instead of executing all transactions till provided one (as was before these changes). I'll fix it, just need to figure out correct signature (maybe I'll provided txHash as a hint when to stop or filter function ... TBD)

Reply:
Yeah, we can just add an index argument to signify how many transactions to run. Can be done in a separate PR

Reply:
Added in this PR